src/lib/core/rxjs/rx-operators.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Observable, ObservableInput} from 'rxjs/Observable';
+import {PartialObserver} from 'rxjs/Observer';
+import {Subscription} from 'rxjs/Subscription';
+import {Scheduler, IScheduler} from 'rxjs/Scheduler';
+import {_finally as _finallyOperator} from 'rxjs/operator/finally';
+import {_catch as _catchOperator} from 'rxjs/operator/catch';
+import {_do as _doOperator} from 'rxjs/operator/do';
+import {map as mapOperator} from 'rxjs/operator/map';
+import {filter as filterOperator} from 'rxjs/operator/filter';
+import {share as shareOperator} from 'rxjs/operator/share';
+import {first as firstOperator} from 'rxjs/operator/first';
+import {switchMap as switchMapOperator} from 'rxjs/operator/switchMap';
+import {startWith as startWithOperator} from 'rxjs/operator/startWith';
+import {debounceTime as debounceTimeOperator} from 'rxjs/operator/debounceTime';
+import {auditTime as auditTimeOperator} from 'rxjs/operator/auditTime';
+import {takeUntil as takeUntilOperator} from 'rxjs/operator/takeUntil';
+
+/**
+ * Represents a strongly-typed chain of RxJS operators.
+ * @docs-private
```

Comment:
I'd expand to something like
```ts
/**
 * Represents a strongly-typed chain of RxJS operators.
 *
 * We achieve strict type enforcement on the chained operators by creating types that
 * *unambiguously* match specific rxjs operators. These unambiguous types are created by
 * intersecting a "brand" to the `typeof` the existing operator. The brand (a class with a private
 * member) effectively forces nominal typing for the operators. This allows typescript to understand
 * that, for example, `filter` is *`filter`* and not, say, a map of T => boolean.
 *
 * The downside to this approach is that operators must be imported in their type-coerced form
 * rather than from the normal rxjs location.
 *
 * @docs-private
 */
```